### PR TITLE
:sparkles: (go/v4) Add GitHub Actions

### PR DESCRIPTION
**Pull request by camilamacedo86**
Within the changes of this PR new projects will become to be scaffold with GitHub actions which will allow users be aware of how to test their changes on the ci.

Either will make clear the purpose of the test-e2e and encorage common/best practices to ensure the quality of the properties

Closes: https://github.com/kubernetes-sigs/kubebuilder/issues/4088

### CODE REVIEW

**Comment by camilamacedo86** [comment]
/hold cancel
It seems fine to fly 🚀

File: pkg/plugins/golang/v4/scaffolds/init.go
Comment (TAM360) [inline pkg/plugins/golang/v4/scaffolds/init.go:166]: @camilamacedo86 Everything looks good to me. Just one question: Is it necessary to maintain separate files for each of GH Action template?

Comment (camilamacedo86) [reply pkg/plugins/golang/v4/scaffolds/init.go:166]: We want to set up 3 separate jobs as follows:

Screenshot 2024-09-03 at 18 53 41


Each job will have its own file so that if a failure occurs, it's easy to identify which target/job/test failed. Running the jobs in parallel will also make the process faster and easier to manage.

This separation also makes it clear which file corresponds to each task: test.yml for tests, lint.yml for linting, and so on. This approach simplifies any future changes and helps maintain clarity—if everything were in the same file, it would be more challenging to pinpoint issues and make updates.

Comment (TAM360) [reply pkg/plugins/golang/v4/scaffolds/init.go:166]: @camilamacedo86  makes sense to maintain the separation of concern. Everything else seems good for now.

**Review by TAM360 — APPROVED** [review]

File: docs/book/src/cronjob-tutorial/testdata/project/.github/workflows/test-e2e.yml
Comment (Kavinjsir) [inline docs/book/src/cronjob-tutorial/testdata/project/.github/workflows/test-e2e.yml:34]: Nit: looks like an additional indent here?

**Review by Kavinjsir — APPROVED** [review]
Looks good to me! 🚀

File: docs/book/src/cronjob-tutorial/testdata/project/.github/workflows/test-e2e.yml
Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/project/.github/workflows/test-e2e.yml:34]: Lets fix this one before get merged

File: docs/book/src/cronjob-tutorial/testdata/project/.github/workflows/lint.yml
Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/project/.github/workflows/lint.yml:23]: I added the same version that we are using.
so we can upgrade all by once

Comment (Kavinjsir) [reply docs/book/src/cronjob-tutorial/testdata/project/.github/workflows/lint.yml:23]: Sounds good!

**Comment by camilamacedo86** [comment]
/hold cancel



Now it is good to fly 🎉

**Pull request merged**
